extension/json/buffered_json_reader.cpp

```diff
@@ -135,9 +138,23 @@ idx_t JSONFileHandle::ReadFromCache(const char *&pointer, idx_t &size, idx_t &po
 		}
 		total_offset += cached_buffer.GetSize();
 	}
+
 	return read_size;
 }
 
+idx_t JSONFileHandle::ReadInternal(const char *pointer, const idx_t requested_size) {
+	// Deal with reading from pipes
+	idx_t total_read_size = 0;
+	while (total_read_size < requested_size) {
+		auto read_size = file_handle->Read((void *)(pointer + total_read_size), requested_size - total_read_size);
```

Comment:
This feels like something that should likely be handled in the `LocalFileSystem::Read` implementation. I imagine this might be causing silent issues in other readers as well?